### PR TITLE
Rename prompt buffer concepts

### PR DESCRIPTION
**Pull request by aadcg**
As discussed in #2234.

### CODE REVIEW

File: libraries/prompter/prompter-source.lisp [hunk 1]
```diff
@@ -382,33 +384,32 @@ If update calculation is aborted, nil is sent instead.")
...
-return the prompter.
-For convenience, it may be initialized with a single function, in which case it
-will be automatically turned into a list."))
+   (auto-selection-action-p nil
+                            :type boolean
+                            :documentation
+                            "Whether the first of `selection-actions' runs automatically.
+Also see `auto-return-selection-action-delay'.")
+
+   (auto-return-selection-action-delay 0.0
```
Comment (Ambrevar) [inline libraries/prompter/prompter-source.lisp:393]: Shouldn't it be `auto-selection-action-delay`?

**Comment by Ambrevar** [comment]
Overall it's much better I believe.

Just one itch: "auto-selection-action" in English sounds like the action on an auto-select, but there is no such thing.  It's actually the "automatic action on selection".  Can we come up with a better term still?

**Comment by jmercouris** [comment]
> Overall it's much better I believe.
> 
> Just one itch: "auto-selection-action" in English sounds like the action on an auto-select, but there is no such thing. It's actually the "automatic action on selection". Can we come up with a better term still?

`automatic-on-selection-action`?

**Comment by Ambrevar** [comment]
Actually I was unclear, it's not actions we are talking about, but the boolean option.
What about this: `auto-action-on-select-p`?  Or `auto-on-select-action-p`?

File: libraries/prompter/prompter-source.lisp (hunk 1 above)
Comment (aadcg) [reply libraries/prompter/prompter-source.lisp:393]: Oh yes, my lack of attention is to blame!

**Comment by aadcg** [comment]
Looks ready to merge to me. @jmercouris @Ambrevar

**Comment by Ambrevar** [comment]
Looks good to me, but I'd like the input of @aartaka on this one.

File: libraries/prompter/prompter-source.lisp [hunk 2]
```diff
@@ -382,33 +384,32 @@ If update calculation is aborted, nil is sent instead.")
...
+Also see `auto-action-on-selection-delay'.")
+
+   (auto-action-on-selection-delay 0.0
+                                   :documentation
+                                   "Execute the first of `selection-actions' after
+this delay when `auto-action-on-selection-p' is non-nil.")
+
+   (selection-actions '()
+                      :type (or null
+                                (or function function-symbol)
```
Comment (aartaka) [inline libraries/prompter/prompter-source.lisp:400]: Oh well, we do really need  a `list-of` type specializer...

File: libraries/prompter/prompter.lisp [hunk 3]
```diff
@@ -157,12 +159,12 @@ compution is not finished.")))
 (defmethod (setf selection) (value (prompter prompter))
   (setf (slot-value prompter 'selection) value)
   (let ((source (selected-source prompter)))
-    (when (follow-p source)
-      (if (< 0 (follow-delay source))
+    (when (auto-action-on-selection-p source)
+      (if (< 0 (auto-action-on-selection-delay source))
           (run-thread "Prompter follow selection thread"
```
Comment (aartaka) [inline libraries/prompter/prompter.lisp:164]: "Prompter auto selection action thread"?

**Review by aartaka — COMMENTED** [review]
My input is: why use `auto-` if we do imply the automatic character of `selection-actions`? `action-on-selection`/`selection-action` sounds like an event handler already, no need for `auto` :)

File: libraries/prompter/prompter-source.lisp [hunk 4]
```diff
@@ -382,33 +384,32 @@ If update calculation is aborted, nil is sent instead.")
...
-                          :type (or null
-                                    (or function function-symbol)
-                                    (cons (or function function-symbol) *))
-                          :documentation
-                          "The first function of this list is called
-automatically on the selection when it's changed.  It does not interrupt or
-return the prompter.
-For convenience, it may be initialized with a single function, in which case it
-will be automatically turned into a list."))
+   (auto-action-on-selection-p nil
```
[the remark this answers is not in the document]
Comment (Ambrevar) [reply libraries/prompter/prompter-source.lisp:387]: I'm not sure how to understand it.  I find it a bit confusing.

File: libraries/prompter/prompter.lisp [hunk 5]
```diff
@@ -188,9 +190,9 @@ Signal destruction by sending a value to PROMPTER's `interrupt-channel'."
   (calispel:! (sync-interrupt-channel (sync-queue prompter)) t)
   (calispel:! (interrupt-channel prompter) t))
 
-(export-always 'call-follow-mode-function)
-(defun call-follow-mode-function (prompter)
-  (sera:and-let* ((action (first (follow-mode-functions (selected-source prompter))))
+(export-always 'call-selection-action)
+(defun call-selection-action (prompter)
+  (sera:and-let* ((action (first (selection-actions (selected-source prompter))))
```
[the remark this answers is not in the document]
Comment (Ambrevar) [reply libraries/prompter/prompter.lisp:195]: Not implemented, I need to finish this part :p

[the remark this answers is not in the document]
Comment (aadcg) [reply libraries/prompter/prompter.lisp:195]: Been there @aartaka, and that was the subject I brought up on issue #2198 :p

No pressure @Ambrevar!

File: libraries/prompter/prompter-source.lisp (hunk 4 above)
[the remark this answers is not in the document]
Comment (aadcg) [reply libraries/prompter/prompter-source.lisp:387]: @aartaka, I agree with you when you say that `auto-` is implied. But I'd prefer `action-on-selection-p` than `selections-actions-p`.

Are you suggesting that `selection-action` and `return-action` could become `action-on-selection` and `action-on-return`?

File: libraries/prompter/prompter.lisp (hunk 3 above)
Comment (aadcg) [reply libraries/prompter/prompter.lisp:164]: Given your comments below, I believe we should leave "auto" out.

File: libraries/prompter/prompter-source.lisp (hunk 2 above)
Comment (aadcg) [reply libraries/prompter/prompter-source.lisp:400]: That would be nice for sure.

File: libraries/prompter/prompter-source.lisp (hunk 4 above)
[the remark this answers is not in the document]
Comment (aartaka) [reply libraries/prompter/prompter-source.lisp:387]: Consistency would't hurt here, so yes, either `actions-on-*` or `*-actions` :)

[the remark this answers is not in the document]
Comment (aadcg) [reply libraries/prompter/prompter-source.lisp:387]: Right, we went for the latter. I was just wondering if you're bent on the former.

**Comment by aadcg** [comment]
If I understood @aartaka right, I believe that the latest version should address all of his comments.

**Comment by aartaka** [comment]
Well, the seeming disconnectedness between `selection-actions` and `action-on-selection-p` still bothers me (maybe call `action-on-selection-(p|delay)` as `selection-actions-(enabled-p|delay)` to reinforce the similarity and make the meaning even clearer? 🤔

**Comment by aadcg** [comment]
Agreed. Either that, or rename `return-actions` and `selection-actions` to `action-on-return` and `action-on-selection`. Any strong opinion?

I like the fact that both would start by `action`...

**Comment by Ambrevar** [comment]
On second thought, I agree with Artyom's proposal.  Let's go with `return-action` and `selection-action`, it's shorter, more consistent, and a glimpse at the doc will make the meaning clear.

**Comment by Ambrevar** [comment]
Please don't merge this before #2253 as it would make a rebase needlessly difficult.

**Comment by aadcg** [comment]
Ok, waiting on #2253.

**Comment by Ambrevar** [comment]
Feel free to rebase and merge!

**Comment by aadcg** [comment]
So, unless I merge from the web UI this is marked as `closed`? I merged from magit, hoping that everything would just work...

**Comment by Ambrevar** [comment]
I don't understand your question.

If you merge from Magit and push to master _without rebasing_, then GitHub detects the PR and closes it automatically.  However if you rebase offline and push to master, GitHub has little way to know which PR is related so it won't do anything.

**Comment by aadcg** [comment]
My point was that if you do it from magit, the github interface shows that the PR wasn't merged but closed (the red icon). When it's done via github, the other icon is shown.

**Comment by Ambrevar** [comment]
Oh sure, but that's not a big deal :)

**Pull request closed without merging**